src/collect-all/Utilities.cs

```diff
@@ -0,0 +1,467 @@
+         proc.OutputDataReceived += (sender, arg) => { var output = arg.Data; };
+         proc.ErrorDataReceived += (sender, arg) => { var err = arg.Data; };
+         
+         proc.Start();
+         proc.BeginOutputReadLine();
+         proc.BeginErrorReadLine();
+
+         proc.WaitForExit();
+         
+         string anyCpuRelease = ".AnyCPU.Release";
+         string osComboName = this.m_os + anyCpuRelease;
+         string osComboNameNative = this.m_os + "." + this.m_arch + ".Release";
+         
+         string coreFxBinDir = Path.Combine(corefxDirectory, "bin", osComboName); 
+         coreFxBinDir += ";" + Path.Combine(corefxDirectory, "bin", "Unix" + anyCpuRelease);
+         coreFxBinDir += ";" + Path.Combine(corefxDirectory, "bin", "AnyOS" + anyCpuRelease);
+         string coreFxNativeBinDir = Path.Combine(corefxDirectory, "bin", osComboNameNative);
+
+         foreach (string directory in coreFxBinDir.Split(new char[1] {';'}))
+         {
+            if (!Directory.Exists(directory))
+            {
+               throw new DirectoryNotFoundException($"{directory} not found.");
+            }
+         }         
+
+         if (!Directory.Exists(coreFxNativeBinDir))
+         {
+            throw new DirectoryNotFoundException($"{coreFxNativeBinDir} not found.");
+         }
+         
+         this.m_corefxBinDir = coreFxBinDir;
+         this.m_nativeCoreFxBinDir = coreFxNativeBinDir;
+      }
+      
+      // Set Core_Root for superPMI.
+     
+      if (this.m_loggingVerbose)
+      {
+         Console.WriteLine(" ");
+         Console.WriteLine($"CORE_ROOT: {this.m_coreRoot}");
+         Console.WriteLine(" ");
+      }
+     
+      Environment.SetEnvironmentVariable("CORE_ROOT", this.m_coreRoot);
+
+   }
+   
+   /////////////////////////////////////////////////////////////////////////////
+   // Public Methods
+   /////////////////////////////////////////////////////////////////////////////
+   
+   public static Utilities GetInstance(string tempPath = null, bool crossGen = false, bool loggingVerbose = false, string arch = null, string buildType = null, string coreclrRepoLocation = null)
+   {
+      if (Utilities.m_Instance == null)
+      {
+         Utilities.m_Instance = new Utilities(tempPath, crossGen, loggingVerbose, arch, buildType, coreclrRepoLocation);
+      }
+      
+      return Utilities.m_Instance;
+   }
+   
+   /////////////////////////////////////////////////////////////////////////////
+   // Utility Methods
+   /////////////////////////////////////////////////////////////////////////////
+   
+   public void CopyDir(string source, string dest)
+   {
+      DirectoryInfo dir = new DirectoryInfo(source);
+      
+      if (!dir.Exists)
+      {
+         string err_msg = $"Source directory {source} does not exit";
+         throw new DirectoryNotFoundException(err_msg);
+      }
+      
+      // If the directory does not exist, create it.
+      if (!Directory.Exists(dest))
```

Comment:
You don't need to check if the dir already exists. If it does, CreateDirectory does nothing.